apps/src/codebridge/Codebridge.tsx

```diff
@@ -54,6 +58,14 @@ export const Codebridge = React.memo(
 
     const projectUtilities = useProjectUtilities(dispatch);
 
+    const currentProjectVersion = useRef(projectVersion);
+    useEffect(() => {
+      if (projectVersion !== currentProjectVersion.current) {
+        projectUtilities.replaceProject(project);
+        currentProjectVersion.current = projectVersion;
+      }
+    }, [currentProjectVersion, project, projectUtilities, projectVersion]);
+
```

Comment:
Keep track of whatever version of the project was handed to us, and if we're given a new projectVersion, then assume that the the external project has changed and replace the internal one.

Yes, it'd be keen not to need to manage two separate bits of data, and hopefully there's a way to do it...